onnxruntime/core/providers/cpu/math/matmul.cc

```diff
@@ -153,8 +128,14 @@ Status MatMul<float>::Compute(OpKernelContext* ctx) const {
   const Tensor* a = ctx->Input<Tensor>(0);
   const Tensor* b = packed_b_ ? nullptr : ctx->Input<Tensor>(1);
 
+  // match CUDA kernel implementation, ignore transpose for vectors
+  const bool trans_a = trans_a_attr_ && a->Shape().NumDimensions() != 1;
+  const bool trans_b = trans_b_attr_ &&
+                       (b == nullptr ? b_shape_.NumDimensions() != 1
```

Comment:
The logic to compute the b_shape pointer should be moved out to a variable so that the logic isn't duplicated here and in the below helper.Compute call.